### PR TITLE
feat: add an env to enable or disable block cache of an app

### PR DESCRIPTION
**Pull request by ZhongChaoqiang**
Sometimes, there are some apps of different business in a cluster.Some apps  want to enable block cache, others my be not.
So we need add an env to enable or disable block cache of an app.

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]

**Comment by neverchanje** [comment]
This feature makes sense. But how do you suppose to close a rocksdb's block cache at runtime after it's created?

**Comment by ZhongChaoqiang** [comment]
> This feature makes sense. But how do you suppose to close a rocksdb's block cache at runtime after it's created? Does rocksdb provide such an API?

We don't need to close rocksdb's block ceche, the cache in rocksdb will be removed by LRU.

**Comment by neverchanje** [comment]
Do you have any graphs or testing results proving that? 

> the cache in rocksdb will be removed by LRU. 
What's the mechanism behind?

**Review by neverchanje — DISMISSED** [review]

**Comment by hycdong** [comment]
Thanks for your contribution~ Cloud you please add an issue in [new_issue_in_pegasus](https://github.com/apache/incubator-pegasus/issues/new/choose)? Besides, as @neverchanje comments, could you please add your test process and explain how RocksDB exchange block-cache-disabled table by LRU and how to promise the block_cache of this table is disabled?

**Comment by ZhongChaoqiang** [comment]
> Do you have any graphs or testing results proving that?
> 
> > the cache in rocksdb will be removed by LRU.
> 
> What's the mechanism behind?

We control it by fill_cache option of rocksdb::ReadOptions.
 If fill_cache  is true, when reading data from rocksdb, the data block will be filled into cache.
If fill_cache  change to false,the data block will not be filled into cache.But we do not need to clear the block of cached.When block cache is full, the oldest block will be remove by LRU.
You can see it in PR of pegasus.(https://github.com/apache/incubator-pegasus/pull/792/files)

File: src/meta/app_env_validator.cpp [hunk 1]
```diff
@@ -161,6 +161,7 @@ void app_env_validator::register_all_validators()
         {replica_envs::ROCKSDB_USAGE_SCENARIO, nullptr},
         {replica_envs::ROCKSDB_CHECKPOINT_RESERVE_MIN_COUNT, nullptr},
         {replica_envs::ROCKSDB_CHECKPOINT_RESERVE_TIME_SECONDS, nullptr},
+        {replica_envs::ROCKSDB_BLOCK_CACHE_ENABLED, nullptr},
```
Comment (levy5307) [inline src/meta/app_env_validator.cpp:164]: could you add a check function for ROCKSDB_BLOCK_CACHE_ENABLED?

**Comment by levy5307** [comment]
And could you add a new issue for this pr?

File: src/meta/app_env_validator.cpp (hunk 1 above)
Comment (ZhongChaoqiang) [reply src/meta/app_env_validator.cpp:177]: > could you add a check function for ROCKSDB_BLOCK_CACHE_ENABLED?

OK，I have added it.

**Comment by ZhongChaoqiang** [comment]
> And could you add a new issue for this pr?

OK! The issue is apache/incubator-pegasus#795

**Comment by ZhongChaoqiang** [comment]
> And could you add a new issue for this pr?

OK! The issue is apache/incubator-pegasus#795

**Review by neverchanje — DISMISSED** [review]

**Review by neverchanje — APPROVED** [review]

**Review by levy5307 — APPROVED** [review]

**Pull request merged**